builds/misc/templates/base-image-publish/base-image-docker-login.yaml

```diff
@@ -0,0 +1,9 @@
+steps:
+- pwsh: |
+    $artifactLocale = "$(Build.SourcesDirectory)/DOCKERIO_PASS"
+    $env:DOCKERIO_PASS | Out-File -Encoding Utf8 $artifactLocale
+    cat $artifactLocale | docker login docker.io --username $(dockerio.username) --password-stdin
+    Remove-Item $artifactLocale -ErrorAction Ignore
+  displayName: Docker login
+  env:
+    DOCKERIO_PASS: $(dockerio.password)
```

Comment:
This is a tricky one. I have tested the login with the Dokcer@2 task but I couldn't figure out how to get the `connection` to go properly given that we are not logging into Azure Container Registry. We are logging into `https://hub.docker.com/r/azureiotedge/` (aka docker.io)

Reply:
You have to create a new Docker service connection in Azure pipelines. In the example I gave earlier, someone created a service connection called "iotedge-edgebuilds-acr" that connects to edgebuilds. You'd have to do the same, only for docker.io. Something like `containerRegistry: docker.io` won't work because there's no service connection called "docker.io" in our Azure pipelines...you have to create one.